src/main/java/org/testng/internal/DynamicGraph.java

```diff
@@ -50,64 +49,103 @@ public void addNode(T node) {
   /**
    * Add an edge between two nodes.
    */
-  public void addEdge(T from, T... tos) {
-    addEdge(from, Arrays.asList(tos));
+  public void addEdge(int weight, T from, T... tos) {
+    addEdge(weight, from, Arrays.asList(tos));
   }
 
   /**
    * Add an edge between two nodes.
    */
-  public void addEdge(T from, Iterable<T> tos) {
+  public void addEdge(int weight, T from, Iterable<T> tos) {
```

Comment:
Ok then forget the overload with a default `weight` of 0 and let's have everything (the methods and `Edge`) take parameters in the order `weight, from, tos`.

What do you think?